libraries/achartengine/achartengine/src/org/achartengine/GraphicalActivity.java

```diff
@@ -15,34 +15,59 @@
  */
 package org.achartengine;
 
-import org.achartengine.chart.AbstractChart;
-
+import android.app.ActionBar;
 import android.app.Activity;
+import android.os.Build;
 import android.os.Bundle;
+import android.view.MenuItem;
 import android.view.Window;
 
+import org.achartengine.chart.AbstractChart;
+
 /**
  * An activity that encapsulates a graphical view of the chart.
  */
 public class GraphicalActivity extends Activity {
-  /** The encapsulated graphical view. */
-  private GraphicalView mView;
-  /** The chart to be drawn. */
-  private AbstractChart mChart;
+    /**
+     * The encapsulated graphical view.
+     */
+    private GraphicalView mView;
+    /**
+     * The chart to be drawn.
+     */
+    private AbstractChart mChart;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Bundle extras = getIntent().getExtras();
+        mChart = (AbstractChart) extras.getSerializable(ChartFactory.CHART);
+        mView = new GraphicalView(this, mChart);
+        String title = extras.getString(ChartFactory.TITLE);
+        if (title == null) {
+            requestWindowFeature(Window.FEATURE_NO_TITLE);
+        } else if (title.length() > 0) {
+            setTitle(title);
+        }
+        setContentView(mView);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            ActionBar actionBar = this.getActionBar();
+            actionBar.setDisplayShowHomeEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+    }
 
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    Bundle extras = getIntent().getExtras();
-    mChart = (AbstractChart) extras.getSerializable(ChartFactory.CHART);
-    mView = new GraphicalView(this, mChart);
-    String title = extras.getString(ChartFactory.TITLE);
-    if (title == null) {
-      requestWindowFeature(Window.FEATURE_NO_TITLE);
-    } else if (title.length() > 0) {
-      setTitle(title);
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:

```

Comment:
I prefer if/else statements in the place of switch statements that only have one or two cases. This is because switch statements have much trickery semantics: easy to fall through to the next case, easy to forget a case, default cases are often awkward or completely left out, etc.